mlflow/crewai/autolog.py

```diff
@@ -0,0 +1,228 @@
+import inspect
+import json
+
+import mlflow
+from mlflow.entities import SpanType
+from mlflow.entities.span import LiveSpan
+from mlflow.tracing.utils import TraceJSONEncoder
+from mlflow.utils.autologging_utils.config import AutoLoggingConfig
+
+
+def patched_class_call(original, self, *args, **kwargs):
+    config = AutoLoggingConfig.init(flavor_name=mlflow.gemini.FLAVOR_NAME)
+
+    if config.log_traces:
+        fullname = f"{self.__class__.__name__}.{original.__name__}"
+        with mlflow.start_span(
+            name=fullname,
+            span_type=_get_span_type(self.__class__, fullname),
+        ) as span:
+            inputs = _construct_full_inputs(original, self, *args, **kwargs)
+            span.set_inputs(inputs)
+            _set_span_attributes(span=span, instance=self)
+            result = original(self, *args, **kwargs)
+            # need to convert the response of generate_content for better visualization
+            outputs = result.__dict__ if hasattr(result, "__dict__") else result
+            span.set_outputs(outputs)
+
+            return result
+
+
+def _get_span_type(cls, task_name: str) -> str:
+    from crewai.flow.flow import Flow
+
+    if issubclass(cls, Flow):
+        return SpanType.CHAIN
+    span_type_mapping = {
```

Comment:
Looks like we don't need method level mapping here? Each class maps to single span type (probably our span types aligns with their class design). We MAY need them but seems YAGNI at this moment.

Reply:
Let me change it to class based approach here.